### ISSUE TITLE
Issues with Main and In-Game menus

### ISSUE THREAD

**Issue by teh53m1chr157** [elishacloud/Silent-Hill-2-Enhancements#88]
Whenever I click on the controls section of the Options menu, it takes around 20 seconds to load the next screen as well as when I exit said menu screen back to Options and subsequently, back to the Main menu. This also happens when opening the Inventory menu or when examining objects, such as Mary's letter at the beginning.

Edit: Apologies for the double-post, my VPN caused some lag.

**Comment by teh53m1chr157** [comment]
Ah! Perfect! Thanks so much. I was hitting a brick wall Googling the subject, lol.

**Comment by teh53m1chr157** [comment]
I am however having trouble getting the D-Pad to function as movement through XInput. The suggestion on the official page is not working. It should be noted that I had previously set the configuration without that option. Do I need to delete dlls before trying again?

**Comment by Polymega** [comment]
Glad to hear! May I ask what your GPU and display resolution is? I'd like to see if there's a common denominator for this. 

Also, I've been dragging my feet on it, but I plan to update ReShade and, if tests are good, share an updated version on the Installation Guide. Maybe the updated version will allow you to use ReShade again. I'll ping you once I have those files ready down the line. 

> It should be noted that I had previously set the configuration without that option. Do I need to delete dlls before trying again?

All you should need to do is open the Xinput Plus software again, set sh2pc.exe as the target file, and follow the steps on the guide to make the d-pad work as a movement pad. You'll need to apply these new changes through the Xinput Plus software (thus, having it overwrite its previous .dll files it makes in the game's directory). 

And, as always, you'll need to relaunch the game any time you adjust external settings to see the changes take effect.

**Comment by Polymega** [comment]
@teh53m1chr157 
Do me a favor and try out this new ReShade package. It's updated to v 4.1.0. Let me know if that resolves the lag/stuttering while still being able to use ReShade.

1. Delete the following files/folders from the game's installation directory, if you haven't already done so: 
   - `\reshade-shaders\`
   - `d3d9.dll`
   - `d3d9.ini`
   - `d3d9.log`
   - `ReShade Filter - SH2 PC PS2.ini`
2. [Download the new ReShade package. ](http://www.enhanced.townofsilenthill.com/SH2/files/SH2PC_ReShade_4.1.0_DirectX9.zip)
   - Place the files in your game's installation directory
3. Ensure `d3d8to9 =` is set to `1` in the `d3d8.ini` file.

**Comment by teh53m1chr157** [comment]
2048MB ATI Radeon RX 560 Series is the GPU I'm running. I had been messing around with my resolution settings, so I'd accidentally left them on 3840x2160, but just now reverted that to 1920x1080 lol. I am using a DS4 as my controller, but I'm using DS4Windows, so for all intents and purposes, digitally it's an Xbox One Controller. I'm gonna try using ReShade on the 1080 settings and see if the 4k had anything to do with it.

**Comment by teh53m1chr157** [comment]
And I'll do that right now.

**Comment by teh53m1chr157** [comment]
Tried the new files and it's reduced the amount of lag that was occurring from 20 seconds to around 7-8 seconds.

**Comment by Polymega** [comment]
Interesting. Your change in resolution is probably what reduced that time then. From the other person who was having issues, it seems _only_ the SMAA filter is causing the issues. So if you really wanted to use ReShade still, you can press Ctrl + W while in-game to open the ReShade control panel, and uncheck the SMAA filter.

**Comment by teh53m1chr157** [comment]
Unchecking SMAA reduced it to about a 4 second delay, so it seems that there's another issue involved.

**Comment by Polymega** [comment]
One more thing, actually: Are you playing in fullscreen mode? (`EnableWndMode =` is set to `0` in the `d3d8.ini` file)

**Comment by teh53m1chr157** [comment]
Yes

**Comment by teh53m1chr157** [comment]
Wait lemme double check, but I believe so

**Comment by teh53m1chr157** [comment]
Yes, that's the case.

**Comment by Polymega** [comment]
This is an issue with AMD cards and the `FastTransitions` mod that's mentioned in the [Troubleshooting portion](http://enhanced.townofsilenthill.com/SH2/#troubleshooting) of the Installation Guide. 


Issue | OS | GPU | Recommendation
-- | -- | -- | --
The FastTransitions feature is noticeably delayed while playing in fullscreen mode. | Any | AMD | Set `EnableWndMode = 1`, `WndModeBorder = 0`, `ResX = 0`, and `ResY = 0` in the `d3d8.ini` file.

In short: Windowed mode must always be on for AMD cards (`EnabledWndMode = 1`). Following the steps above, the game will play in windowed mode to alleviate your issue but still take up the entire display of your screen. You could probably re-enabled SMAA in ReShade if the suggestion above fixes your delay.

**Comment by teh53m1chr157** [comment]
Ah, there it is. You rock, my friend. Thanks for bearing with me, especially considering I had skimmed the Troubleshooting guide, lmao. I'm gonna chalk the Dpad thing up to the buttons being numbered differently on a DS4 as opposed to the Xbox controller, as it's not that big of a deal anyway. Thanks muchly, man!

**Comment by Polymega** [comment]
No problem! I won't be able to assist with DS4Windows as I've never used it but our friend @seagull uses the software and has a sequence of steps to get everything going. Perhaps I can pester him to find the previous post of his mentioning the steps he's done?

**Comment by teh53m1chr157** [comment]
That'd be great! Let me know what you guys find out. I'm having a blast playing through this again. It made a huge impact on me when I was younger lol.

**Comment by seagull** [comment]
Hello, I'm seagull.
Here's what I've always done:

- Install both DS4Windows and a joypad-to-keyboard tool like XPadder or JoyToKey
- Set up DS4Windows so it treats the controller as an Xbox controller
- When mapping to JoyToKey/XPadder, leave the left analogue stick alone as the game understands what that is natively and delivers analogue control from it (otherwise you're stuck with digital)
- Ensure the game has all joypad input options cleared in the options menu or you'll get double inputs
- For extra fun, install the [Lavendy xi driver](http://lavendy.net/special/driver/xi/index.html) for force feedback

Cheers

**Comment by micatius** [comment]
I have an AMD card and tried every possible thing suggested both in this thread and the troubleshooting guide and I still get around 3-4 seconds of loading time on every level or menu transition. Other than that, everything else works perfectly. Tried with and without ReShade, disabled the AA in ReShade, tried windowed mode, tried disabling Fast Transitions, the loading time remained constant.

**Comment by Polymega** [comment]
> Tried with and without ReShade, disabled the AA in ReShade, tried windowed mode, tried disabling Fast Transitions, the loading time remained constant.

That would have been everything I'd suggest to try. Are you by chance using a fairly old laptop or the like for the game? Are you running the game at a crazy-high resolution?

Perhaps try disabling every fix, one-by-one, to see if that alleviates the problem. If you have an AMD card running the game, windowed mode should've fixed that problem. Hmm...

**Comment by micatius** [comment]
Okay, my bad. I was using notepad++ for the first time and didn't realize I have to click the save icon instead of just exiting the app for the changes to take place. What confused me is when you open the file again in notepad++ the changes will show but won't apply until you actually press the diskette icon. Thanks for help though Polymega

**Comment by Polymega** [comment]
Good deal.

>What confused me is when you open the file again in notepad++ the changes will show but won't apply until you actually press the diskette icon.

You can disable that and have it work like any other Word editor which is something I've also done to avoid confusion for that very same reason!

**Comment by elishacloud** [comment]
Closing issue.

**Comment by AeroWidescreen** [comment]
Hello, if you're still having trouble with ReShade then please try this new build of SH2E. It includes post-processing effects that are designed to be more efficient than ReShade and resolve any performance issues you may be having.

Verify that ReShade has been _removed_ before attempting to try this. Settings can be adjusted in the **[SHADERS]** section of the INI file.

[d3d8_b1.5.1806.zip](https://mega.nz/file/dP5QCbQK#kOX7UST9b2LeAhWiN4d7zuS9fJ0JmU6ZTljGJktsQiI)
